routing/router.go

```diff
@@ -698,18 +706,19 @@ func (r *ChannelRouter) FindBlindedPaths(destination route.Vertex,
 		return routes[i].probability > routes[j].probability
 	})
 
-	// Now just choose the best paths up until the maximum number of allowed
-	// paths.
-	bestRoutes := make([]*route.Route, 0, restrictions.MaxNumPaths)
+	log.Debugf("Found %v routes, discarded %v low probability routes",
+		totalRoutesFound, totalRoutesDiscarded)
+
+	// Return all routes, capped by the maxNumberOfRoutes.
+	allRoutes := make([]*route.Route, 0, len(routes))
 	for _, route := range routes {
-		if len(bestRoutes) >= int(restrictions.MaxNumPaths) {
+		if len(allRoutes) >= maxRoutes {
```

Comment:
that is not what I meant, my idea was to exit the recursion early when enough paths are found,

I was more thinking of restricting the recursion in `processNodeForBlindedPath` but I would like to get @ellemouton opinion on this and also this can probably done in another PR. 

Currently our DFS exhausts all Paths which exist in the graph, so I was thinking we could abort the DFS after we already have a `hopSet` of let's say 100 ?

Imagine we have a minRouteLen of 2 and a maxRouteLen of 6, each node has more than 100 channels, when reaching the depth of 6, we could easily accumulate > 1000 routes which seems unreasonable.

But the above restrictions we do not need, because really makes not difference we already have all the paths in memory.